connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConfig.java

```diff
@@ -101,7 +101,11 @@ public class MirrorSourceConfig extends MirrorConnectorConfig {
     public static final Class<?> CONFIG_PROPERTY_FILTER_CLASS_DEFAULT = DefaultConfigPropertyFilter.class;
 
     public static final String OFFSET_LAG_MAX = "offset.lag.max";
-    private static final String OFFSET_LAG_MAX_DOC = "How out-of-sync a remote partition can be before it is resynced.";
+    private static final String OFFSET_LAG_MAX_DOC = "Determines the maximum allowed lag between the source and target partitions before MirrorMaker emits an 'offset sync' event to catch up the remote partition. The lag is calculated as the difference between the latest offset in the source partition, and the last sync emitted to the offset syncs topic. " +
+            "When the lag for a remote partition exceeds the <code>offset.lag.max</code> value, MirrorMaker emits an 'offset sync' event to the 'offset-syncs' topic, which is then used by the <code>MirrorCheckpointTask</code> to translate source and target partitions." +
+            "Setting <code>offset.lag.max</code> to a lower value can be beneficial in scenarios where partitions have consistently low throughput, and the fixed sync interval ('offset.flush.interval.ms') is not acceptable. A lower value will trigger more frequent offset syncs, keeping the target partitions more closely in sync with the source for these low-throughput partitions. " +
+            "On the other hand, setting it to a higher value can be useful when the source topic has high throughput and the remote partitions can tolerate a larger lag without impacting target costumers." +
+            "It's also possible to set <code>offset.lag.max</code> to 0, which will cause MirrorMaker to emit an 'offset sync' for every source record  to a target partition. This can be useful for strict synchronization requirements but may increase the load and extra throughput on the 'offset-syncs' topic, depending on the value of the 'offset-syncs.topic.location' configuration";
```

Comment:
Don't use "strict synchronization" here, as it could be confused for "synchronous", which is a capability that MM2 does not have.

Also the mention of the offset-syncs.topic.location needs to be rephrased. If you say "X may do Y, depending on the value of Z", it makes me think there is some value of Z avoids Y, when that isn't the case.

Lowering the value _will_ increase the load on the offset-syncs topic except in very specific scenarios, so using _may_ is more confusing than it's worth.